src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/ObjectController.cs

```diff
@@ -0,0 +1,319 @@
+        /// <summary>
+        ///     The CID of the object.
+        /// </summary>
+        public string Hash { set; get; }
+
+        /// <summary>
+        ///     The object size.
+        /// </summary>
+        public long Size { set; get; }
+    }
+
+    /// <summary>
+    ///     Link details on an object.
+    /// </summary>
+    public class ObjectLinkDetailDto
+    {
+        /// <summary>
+        ///     The CID of the object.
+        /// </summary>
+        public string Hash { set; get; }
+
+        /// <summary>
+        ///     Links to other objects.
+        /// </summary>
+        public IEnumerable<ObjectLinkDto> Links { set; get; }
+    }
+
+    /// <summary>
+    ///     Data and link details on an object.
+    /// </summary>
+    public class ObjectDataDetailDto : ObjectLinkDetailDto
+    {
+        /// <summary>
+        ///     The object data encoded as UTF-8.
+        /// </summary>
+        public string Data { set; get; }
+    }
+
+    /// <summary>
+    ///     Manages the IPFS Merkle Directed Acrylic Graph.
+    /// </summary>
+    /// <remarks>
+    ///     <note>
+    ///         This is being obsoleted by <see cref="IDagApi" />.
+    ///     </note>
+    /// </remarks>
+    public sealed class ObjectController : IpfsController
+    {
+        /// <summary>
+        ///     Creates a new controller.
+        /// </summary>
+        public ObjectController(ICoreApi dfs) : base(dfs) { }
+
+        /// <summary>
+        ///     Create an object from a template.
+        /// </summary>
+        /// <param name="arg">
+        ///     Template name. Must be "unixfs-dir".
+        /// </param>
+        [HttpGet] [HttpPost] [Route("object/new")]
+        private async Task<ObjectLinkDetailDto> Create(string arg)
+        {
+            var node = await IpfsCore.ObjectApi.NewAsync(arg, Cancel);
+            Immutable();
+            return new ObjectLinkDetailDto
+            {
+                Hash = node.Id,
+                Links = node.Links.Select(link => new ObjectLinkDto
+                {
+                    Hash = link.Id,
+                    Name = link.Name,
+                    Size = link.Size
+                })
+            };
+        }
+
+        /// <summary>
+        ///     Store a MerkleDAG node.
+        /// </summary>
+        /// <param name="file">
+        ///     multipart/form-data.
+        /// </param>
+        /// <param name="inputenc">
+        ///     "protobuf" or "json"
+        /// </param>
+        /// <param name="datafieldenc">
+        ///     "text" or "base64"
+        /// </param>
+        /// <param name="pin">
+        ///     Pin the object.
+        /// </param>
+        /// <returns></returns>
+        [HttpPost("object/put")]
+        private async Task<ObjectLinkDetailDto> Put(IFormFile file,
+            string inputenc = "json",
+            string datafieldenc = "text",
+            bool pin = false)
+        {
+            if (datafieldenc != "text") // TODO
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)